cpp/tests/groupby/group_argmax_test.cpp

```diff
@@ -160,8 +161,56 @@ TEST_F(groupby_argmax_string_test, zero_valid_values)
     auto agg2 = cudf::make_argmax_aggregation();
     test_single_agg(keys, vals, expect_keys, expect_vals, std::move(agg2), force_use_sort_impl::YES);
 }
+
 // clang-format on
 
+struct groupby_dictionary_argmax_test : public cudf::test::BaseFixture {
+};
+
+TEST_F(groupby_dictionary_argmax_test, basic)
+{
+  using K = int32_t;
+  using V = string_view;
+  using R = cudf::detail::target_type_t<V, aggregation::ARGMAX>;
+
+  // clang-format off
+  fixed_width_column_wrapper<K> keys_w{     1,     2,    3,     1,     2,     2,     1,    3,    3,    2 };
+  strings_column_wrapper        vals_w{ "año", "bit", "₹1", "aaa", "zit", "bat", "aab", "$1", "€1", "wut"};
+  fixed_width_column_wrapper<K> expect_keys_w{ 1, 2, 3 };
+  fixed_width_column_wrapper<R> expect_vals( { 0, 4, 2 });
+  // clang-format on
+
+  auto keys        = cudf::dictionary::encode(keys_w);
+  auto vals        = cudf::dictionary::encode(vals_w);
+  auto expect_keys = cudf::dictionary::encode(expect_keys_w);
+
+  test_single_agg(
+    keys->view(), vals_w, expect_keys->view(), expect_vals, cudf::make_argmax_aggregation());
+  test_single_agg(
+    keys_w, vals->view(), expect_keys_w, expect_vals, cudf::make_argmax_aggregation());
+  test_single_agg(
+    keys->view(), vals->view(), expect_keys->view(), expect_vals, cudf::make_argmax_aggregation());
+
+  test_single_agg(keys->view(),
```

Comment:
Because the implementation separates the grouping of keys from the aggregation calculation, the test for grouping by key type is in `groupby_keys_test.cpp`. You wouldn't need to test for dictionary keys everywhere.

Reply:
I don't think I understand. Each groupby/agg seemed to require some tweaks to the code in order to support dictionary types. Some cases only needed the indices while others required using the keys. Adding tests for each groupby/agg seemed  prudent to ensure groupby can support dictionary columns. The groupby code is very complex so I may have misunderstanding of how it works.

Reply:
Yes. There's two scenarios: 
1. the keys are dictionary columns
2. the values are dictionary columns
You only need to test for the latter in each of the test files that follow the naming pattern `group_<agg>_test.cpp`. For the former, it is sufficient to add dictionary keys test to `groupby_keys_test.cpp`.